microdnf/commands/repoquery/repoquery.cpp

```diff
@@ -170,25 +170,21 @@ void CmdRepoquery::run(Context & ctx) {
         }
     }
 
-    std::vector<std::string> patterns_to_show;
-    if (patterns_to_show_options->size() > 0) {
-        patterns_to_show.reserve(patterns_to_show_options->size());
-        for (auto & pattern : *patterns_to_show_options) {
-            patterns_to_show.emplace_back(dynamic_cast<libdnf::OptionString *>(pattern.get())->get_value());
-        }
+    libdnf::rpm::PackageSet result_pset(&solv_sack);
+    libdnf::rpm::SolvQuery full_solv_query(&solv_sack);
+    for (auto & pattern : *patterns_to_show_options) {
+        libdnf::rpm::SolvQuery solv_query(full_solv_query);
+        solv_query.resolve_pkg_spec(dynamic_cast<libdnf::OptionString *>(pattern.get())->get_value(), true, true, true, true, true, {});
```

Comment:
Because the performance optimization in dnf-5 is applied only without ICASE, therefore microdnf and dnf should provide a similar performance.